### DISCUSSION TITLE
Question on engine-Q

### DISCUSSION THREAD

**Discussion by 0323pin** [nushell/nushell#4717] [Q&A] [answered]
Hi,

NetBSD package maintainer and `nu` user here.
I've been looking at building and testing `nu` from git-HEAD, kind of curious to see what to expect ahead of the coming release.

Although, I think I can easily fix my issues locally, which I haven't attempted yet, there's something that could become an issue with packaging the coming release. Namely, `reedline`.

As is right now, both `nu-cli` and `nu-command` want `reedline = { git = "https://github.com/nushell/reedline", branch = "main" }`.
Are you planning on pushing a proper release of `reedline` to crates.io ahead of the next release of `nu`? Internally at our end, we can not pull distfiles at compile time and this will require a rather tedious work around.

Thanks for any insights on this.
Regards

**Comment by sophiajt** [comment, marked as answer]
@0323pin - yeah, we're planning to publish a reedline to crates.io soon, before the official 0.60 release. Would it be easier on you if we did it sooner rather than waiting closer to the 0.60 release?

**Reply by 0323pin** [reply]
Thanks! It's good to know your are planning to do that :)

Yes, it would be nice to test it before the real release comes out, just in case something else fails to build on NetBSD.
It would be much simpler to have a `reedline` release to do a test build and prepare the package for the coming release. As a user, I could then test functionality and configuration differences.

Not a must but, really helpful.

**Comment by elferherrera** [comment]
I only want to add the history bang which will be done this weekend

**Reply by 0323pin** [reply]
That sounds like a reasonable waiting time 😄 
I'll look at building `nu` next week.
Gracias!

**Comment by 0323pin** [comment]
Would it be possible to fix `reedline` a week or so before the actual release of the next version of `nu`?

I'd appreciate testing the new engine-Q and it's implications ahead of updating the package at our end.

TIA

**Comment by elferherrera** [comment]
I think reedline is ready for a release. I'm not going to add anything for the time being. @sholderbach @jntrnr what do you think?

**Comment by sholderbach** [comment]
@0323pin https://github.com/nushell/nushell/pull/4850 should solve that

**Comment by 0323pin** [comment]
Thanks!
I was able to start a build but, unfortunately it fails. Will open an issue for it.
